doc/tutorials/highlevel.md

```diff
@@ -0,0 +1,322 @@
+
+Of course you can also use `elektraSet` (error handling omitted):
+
+```c
+elektraSet (elektra, ELEKTRA_TAG_MYDOUBLE, 3.141593, &error);
+elektraSetV (elektra, ELEKTRA_TAG_MYFLOATARRAY, 2.718282f, &error, 2);
+```
+
+Note that `elektraSetV` takes the `ElektraError` argument before the variable arguments, while in `ELEKTA_SET` the error is always the last
+argument. This is simply because of limitations in the C macro system.
+
+There is not setter for array sizes. Since Elektra's low-level part supports discontinuous arrays, we simply change the array size whenever
+necessary, if an array element setter is called. However, the high-level API has no support for discontinuous arrays, so take care not to
+create holes in your arrays, if you want to iterate over them. Remember, accessing non-existent keys (and this includes array elements) is a
+fatal error.
+
+### Advanced concepts
+
+The code-generator has some more advanced features that are supported out of the box. For example, you can read multiple config values at
+once by utilising structs. The use of structs also allows for recursive configurations like menus (a menu can have submenus).
+
+For more information take a look at the man-page [`kdb-gen-highlevel(1)`](../help/kdb-gen-highlevel.md).
+
+## Compiling your application
+
+Once you've written your application, you will want to compile it. This requires linking some libraries and adding to your include path.
+The easiest way is to use CMake or pkg-config to find the needed compiler options. Examples on how set this up can be found in
+[here](/examples/codegen/econf) and [here](/examples/codegen/tree).
+
+The compiler invocation should look something like this:
+
+```sh
+cc myapp.c conf.c `pkg-config --cflags --libs elektra-codegen` -I. -o myapp -Wl,-rpath `pkg-config --variable=libdir elektra-codegen`
+```
+
+Note: At least C99 is required, so if your compiler defaults to an older version you'll need to add `-std=c99`.
+
+## Running your application
+
+Running your application is easy, just run the executable (e.g. `myapp`). While this might work out of the box, you will just get the default
+configuration. To change the configuration you need to use `kdb`, which doesn't know about your specification yet, so you would need to set
+the `type` metadata and all the other stuff that your application expects yourself. For every single key.
+
+### Mounting the specification
+
+A better solution is to inform Elektra (and `kdb`) about our specification. Then Elektra automatically copies metadata to where it should be.
+
+First you need to mount your specification itself into the KDB. Mounting is basically the process of informing Elektra about a new part of
+the KDB, similar to how mounting an external hard drive informs the OS about a new part of the file system.
+
+```sh
+sudo kdb mount -R noresolver myapp_spec.eqd "spec/sw/example/myapp/#0/current" specload app="$PWD/myapp"
```

Comment:
`myapp_spec.eqd` is the override file that contains the changes made by the user and validated by `specload`. The line above would create a file in the current working directory whenever you call `kdb set spec/sw/example/myapp/#0/current/...`.

This line is obviously far from ideal. In a perfect world it would be `sudo kdb mount myapp_spec.eqd "spec/sw/example/myapp/#0/current" specload app="$PWD/myapp"`. But there are some problems in `specload`, if a real resolver is used. Mainly that `specload` won't be called, because the resolver reports no changes.

If you want to make changes to the spec via `specload`, you should provide an absolute path on the system level (e.g. `/etc/myap_spec.eqd`), so that we always access the same file.

Reply:
Yes, maybe we should give it a full path and write down that this file will be used for user-changes.